theurgy.lic

```diff
@@ -55,75 +122,359 @@ def buy_wine
-waitrt?
-DRC.bput('pray meraud', 'meraud')
-waitrt?
+def reflectory
+  DRCT.walk_to '5988'
+  fput 'meditate'
+  waitrt?
+  DRCT.walk_to '1030'
+end
 
-fput 'remove badge'
-fput 'pray badge'
-pause 2
-waitrt?
-fput 'wear badge', 'You put on a', 'You are already'
+def invoke_parchment
+  DRCT.walk_to('1230')
+  if 'but there is nothing in there like that' != DRC.bput("rummage /C parchment my #{$settings.theurgy_supply_container}", 'but there is nothing in there like that', 'pale golden prayer parchment')
+    get_from_container('golden parchment')
+    fput 'invoke my parchment'
+    waitfor 'You conclude'
+    put_in_container('golden parchment')
+  end
+end
 
-until ['flawless performance to those on high', 'In your condition'].include? DRC.bput('dance', 'flawless performance to those on high', 'You begin to dance', 'Your actions grow', 'Your dance', 'but you falt', 'In your condition')
-  pause 1
+######################## CUSTOM TRAVEL ########################
+
+def enter_altar
+  DRCT.walk_to('1164')
+  move 'go shack'
+  move 'go hole'
+  move 'west'
+  move 'up'
   waitrt?
-  DRC.fix_standing
+  move 'go path'
 end
 
-fput "get #{$settings.flint_lighter}", 'You get a', 'You are already'
-get_from_container('incense')
-while 'nothing happens' == DRC.bput('light my incense with flint', 'nothing happens', 'bursts into flames')
+def exit_altar
+  move 'go path'
+  move 'go hole'
   waitrt?
+  move 'east'
+  move 'go hole'
+  move 'out'
 end
+
+######################## MISC ########################
+
+def research_check
+  case find_current_project = DRC.bput("research status", / (\d+)% complete with a portion .*\./, 'not researching anything', 'project about')
+  when /not researching anything/
+    return
+  when /project about/
+    research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+    find_research = research_type.find{ |find_research| find_current_project =~ /#{find_research}/i }
+	$hold_research = find_research
+    if $hold_research.nil?
+      return
+    end
+  when /complete with a portion/
+    echo find_current_project
+    research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+    find_research = research_type.find{ |find_research| find_current_project =~ /#{find_research}/i }
+	$hold_research = find_research
+    if $hold_research.nil?
+      return
+    end
+    if find_current_project[1,2].to_i >= 60
+      wait_time = (101 - find_current_project[1,2].to_i) * 3
+      Flags.reset('research_done')
+      pause (wait_time)
+    else
+      echo "stopping research"
+    end
+    fput 'prep bless'
+  end
+  echo $hold_research
+  return $hold_research
+end
+
+def commune_check
+  echo Flags['theurgy-commune']
+  if Flags['theurgy-commune'] == false
+    next_commune
+  elsif Flags['theurgy-commune'].include? "fully prepared to seek assistance from the Immortals once again"
```

Comment:
try as I might I can't get these to work together as one (return unless... or statement) so set it up this way for now since it works.  Willing to change to a nicer format but I just needed something stable.